tests/Unit/Elliptic/DiscontinuousGalerkin/Test_ImposeBoundaryConditions.cpp

```diff
@@ -0,0 +1,491 @@
+      get<Tags::NormalDotFlux<field_tag>>(flux_vars) =
+          data.exterior_bdry_fluxes.at(direction);
+    }
+
+    db::item_type<other_data_tag> exterior_bdry_other_data;
+    for (const auto& direction : external_directions) {
+      auto& other_data_vars = exterior_bdry_other_data[direction];
+      other_data_vars.initialize(3);
+      get<OtherData>(other_data_vars) =
+          data.exterior_bdry_other_data.at(direction);
+    }
+
+    db::item_type<exterior_bdry_vars_tag> exterior_bdry_vars;
+    for (const auto& direction : external_directions) {
+      auto& vars = exterior_bdry_vars[direction];
+      vars.initialize(3);
+      get<field_tag>(vars) = data.exterior_bdry_vars.at(direction);
+    }
+
+    const int initial_temporal_id = 0;
+    const int next_temporal_id = 1;
+
+    db::item_type<mortar_data_tag<flux_comm_types>> mortar_history{};
+    db::item_type<mortar_next_temporal_ids_tag> mortar_next_temporal_ids{};
+    db::item_type<mortar_meshes_tag> mortar_meshes{};
+    db::item_type<mortar_sizes_tag> mortar_sizes{};
+    for (const auto& mortar_id : external_mortar_ids) {
+      mortar_history.insert({mortar_id, {}});
+      mortar_next_temporal_ids.insert({mortar_id, initial_temporal_id});
+      mortar_meshes.insert({mortar_id, mesh.slice_away(0)});
+      mortar_sizes.insert({mortar_id, {{Spectral::MortarSize::Full}}});
+    }
+
+    return db::create<
+        db::AddSimpleTags<
+            TemporalId, Tags::Next<TemporalId>, Tags::Mesh<2>, Tags::Element<2>,
+            Tags::ElementMap<2>, bdry_vars_tag,
+            bdry_normal_dot_fluxes_tag<flux_comm_types>, bdry_other_data_tag,
+            exterior_bdry_normal_dot_fluxes_tag<flux_comm_types>,
+            exterior_bdry_other_data_tag, exterior_bdry_vars_tag,
+            mortar_data_tag<flux_comm_types>, mortar_next_temporal_ids_tag,
+            mortar_meshes_tag, mortar_sizes_tag>,
+        compute_items<my_component>>(
+        initial_temporal_id, next_temporal_id, mesh, element,
+        std::move(element_map), std::move(bdry_vars),
+        std::move(bdry_normal_dot_fluxes), std::move(bdry_other_data),
+        std::move(exterior_bdry_normal_dot_fluxes),
+        std::move(exterior_bdry_other_data), std::move(exterior_bdry_vars),
+        std::move(mortar_history), std::move(mortar_next_temporal_ids),
+        std::move(mortar_meshes), std::move(mortar_sizes));
+  }
+  ();
+
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+  using MockDistributedObjectsTag =
+      typename MockRuntimeSystem::template MockDistributedObjectsTag<
+          my_component>;
+  typename MockRuntimeSystem::TupleOfMockDistributedObjects dist_objects{};
+  tuples::get<MockDistributedObjectsTag>(dist_objects)
+      .emplace(self_id, std::move(start_box));
+
+  ActionTesting::MockRuntimeSystem<Metavariables> runner{
+      {NumericalFlux{}}, std::move(dist_objects)};
+
+  using initial_databox_type = db::compute_databox_type<tmpl::append<
```

Comment:
Probably clearer and easier to maintain if this is written as `decltype(start_box)`.